### PR TITLE
Toggle component extra dropdown

### PR DESCRIPTION
**Pull request by discodavey**

### CODE REVIEW

File: src/ui/styles/ContentToggle.scss
Comment (NuclearRedeye) [inline src/ui/styles/ContentToggle.scss:6]: These classnames should be renamed to reflect the new component name.

File: src/ui/atoms/ContentToggle.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.tsx:22]: You shouldn't need the additional inner `<span>` tags here.

File: src/ui/atoms/ContentToggle.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.tsx:15]: Missing `id` attribute.

File: src/ui/atoms/ContentToggle.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.tsx:8]: This isn't used nor needed.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:70]: This isn't actually testing what you think it is, '.select-toggle' is always visible. It's the `children` that can be visible or hidden.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:61]: This isn't actually testing what you think it is, '.select-toggle' is always visible. It's the `children` that can be visible or hidden.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:52]: This isn't actually testing what you think it is, '.select-toggle' is always visible. It's the `children` that can be visible or hidden.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]

File: src/ui/styles/ContentToggle.scss
Comment (NuclearRedeye) [inline src/ui/styles/ContentToggle.scss:22]: I can see this isn't used here, but likely will be in a later task though. I suspect though that the css here should be specific to the component and not include styles for other parts hence I think it should be removed.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]
Almost there, just a couple more tweaks

File: src/ui/styles/ContentToggle.scss
Comment (NuclearRedeye) [inline src/ui/styles/ContentToggle.scss:22]: I still can't see this class being used by this component. It should either be moved to another file that does or will use it or be deleted from here.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]

File: src/ui/atoms/ContentToggle.test.tsx
Comment (Cooryd) [inline src/ui/atoms/ContentToggle.test.tsx:14]: Because `children` is an optional prop and this test case should be testing the render of minimal props passed to the component, we can remove the text content here and make this a self closing tag.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (Cooryd) [inline src/ui/atoms/ContentToggle.test.tsx:67]: We should add two more test cases here, one for toggling from each initial state. open > closed and closed > open.

You can use the `fireEvent` object from `'@testing-library/react'` to trigger a click by calling `fireEvent.click(${someSelectorForOnClickElement})`

File: src/ui/styles/ContentToggle.scss
Comment (discodavey) [reply src/ui/styles/ContentToggle.scss:22]: This was being amended from the default height but I have now moved it all to the select-field css so it is all in one place. If you look in the console it did show that there was 0.75rem in the select-field and 0.75rem in the select-field__input to make the 1.5rem (24px) that Chris had in his design. This is now all in .select-field and this may need to be changed when we see it in the form.

Screenshot 2021-05-27 at 08 17 15

Comment (NuclearRedeye) [reply src/ui/styles/ContentToggle.scss:22]: I can see that, but what I was trying to convey is that this specific style class is not a part of this component but rather is to do with the usage of the component. Recall that the goal is to create a generic toggle component, which can have any type or number of sub components. It just so happens that our first usage of the component will be to wrap a SelectField component. Hence these style changes should come later.
Also, by adding that class are you not changing the behaviour of all the existing SelectField components in Reviewer? That could have unintended side effects that we don't want to introduce with this commit.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:79]: Have a re-read of the 2 test cases that @Cooryd asked for. You've covered one of the tests, but this second test isn't correct and should be testing a different user interaction. 

As an aside, this second test here isn't actually testing anything and there is no way it would ever fail.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]

File: src/ui/atoms/ContentToggle.test.tsx
Comment (discodavey) [reply src/ui/atoms/ContentToggle.test.tsx:79]: The second test when I added the Selectfield in with the toBeNull it failed but passed when I took it out.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:46]: This would be better as `should not show content if collapsed`.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:55]: Likewise, `should show content if expanded`.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:64]: And,`should expand when clicked `.

File: src/ui/atoms/ContentToggle.test.tsx
Comment (NuclearRedeye) [inline src/ui/atoms/ContentToggle.test.tsx:79]: And,`should collapse when clicked`.

**Review by NuclearRedeye — COMMENTED** [review]
Some minor changes to the test titles I think you should make, and also a linting error that needs to be cleaned up. But otherwise I think this is ready. Will wait for @Cooryd to give it a final pass next week before merging

File: src/ui/styles/ContentToggle.scss
Comment (NuclearRedeye) [inline src/ui/styles/ContentToggle.scss:22]: Last thing, just remove this class as it isn't used.

**Review by NuclearRedeye — COMMENTED** [review]
Just one minor item to address, but this should be good now and ready for a second review from @Cooryd

**Review by Cooryd — APPROVED** [review]
👍

**Review by NuclearRedeye — APPROVED** [review]

**Pull request merged**
